Sources/TuistSupport/System/Systeming.swift

```diff
@@ -0,0 +1,91 @@
+import Combine
+import Foundation
+
+public protocol Systeming {
+    /// System environment.
+    var env: [String: String] { get }
+
+    /// Runs a command without collecting output nor printing anything.
+    ///
+    /// - Parameter arguments: Command.
+    /// - Throws: An error if the command fails
+    func run(_ arguments: [String]) throws
```

Comment:
I think the `String…` parameter was nicer to use.
What about adding extensions for these which will map to the `[String]` one?

Reply:
I would prefer if we can stick with one, I don't mind if the `[String]` or `String...`: this file was complex because there was a lot of redundant declarations and unused methods. :/

Reply:
Agree, the `Systeming` API in Tuist is not my favorite tbh. We should look to replace the reactive/combine part of this from this API entirely now that `async/await` is a thing it much better describes the type of work a CLI does.

I've been (slowly) dabbling with this myself and have a really rough working example: https://github.com/luispadron/Beans/blob/main/Sources/System/System.swift